core/src/main/java/com/linecorp/armeria/client/ClientRequestContextCustomizers.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+final class ClientRequestContextCustomizers {
+
+    @Nullable
+    private ArrayList<Consumer<? super ClientRequestContext>> customizers;
+
+    private boolean pendingContextCapture;
```

Comment:
I think customizers and context capture don't seem related - do we use the same object to reduce some object allocations? If so, can probably document that.

Reply:
Realized maybe it's just the class name, `ClientThreadLocalState` so it doesn't apply a semantic concept like customization only?

Reply:
Done